### PR TITLE
IPROTO-279 Add identifiers to IllegalStateException thrown for a non-…

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/IPROTO-279

### CODE REVIEW

File: core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionTypeFactory.java [hunk 1]
```diff
@@ -474,7 +474,7 @@ public XClass determineRepeatedElementType() {
             }
             return fromClass(c);
          }
-         throw new IllegalStateException("Not a repeatable field");
+         throw new IllegalStateException(String.format("Not a repeatable field: %s#%s", declaringClass.getName(), method.getName()));
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionTypeFactory.java:477]: Add this to `org.infinispan.protostream.impl.Log` and use it everywhere

**Review by tristantarrant — CHANGES_REQUESTED** [review]

**Comment by ryanemerson** [comment]
Updated.

**Comment by tristantarrant** [comment]
Much better :)

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**